packages/react-jss/src/JssProvider.js

```diff
@@ -67,13 +67,18 @@ export default class JssProvider extends Component<Props> {
       sheetOptions.generateId = generateId
     } else if (!sheetOptions.generateId) {
       if (!this.generateId) {
-        let createGenerateId = createGenerateIdDefault
-        if (localJss && localJss.options.createGenerateId) {
-          createGenerateId = localJss.options.createGenerateId
+        if (localJss && localJss.generateClassName) {
+          // re-use the same classname generator pinned to the local jss instance
+          this.generateClassName = localJss.generateClassName
+        } else {
+          let createGenerateId = createGenerateIdDefault
+          if (localJss && localJss.options.createGenerateId) {
+            createGenerateId = localJss.options.createGenerateId
```

Comment:
I don't think we need this anymore. When a local jss instance is passed, it will always have a `generateId` and this check should be redundant.

Reply:
@kof what was the reason for this? Can't seem to have a use case where it makes sense? I think if the user passes in a jss instance, we should use the `generateId` instead of calling the `createGenerateId` again, as this might lead to duplicate classes.

Reply:
I think we didn't have a cached generateId() on the jss instance back then, try to remove and see if any tests indicate a problem